stdlib/args.ysh

```diff
@@ -3,50 +3,50 @@
 # Usage:
 #   source --builtin args.sh
 #
-# Args :spec {
-#   flag -v --verbose "Verbosely"  # default is Bool, false
+# parser (&spec) {
+#   flag -v --verbose (help="Verbosely")  # default is Bool, false
 #
-#   flag -P --max-procs '''
+#   flag -P --max-procs ('int', default=-1, doc='''
 #     Run at most P processes at a time
-#     ''' (Int, default=-1)
+#     ''')
 #
-#   flag -i --invert '''
+#   flag -i --invert ('bool', default=true, doc='''
 #     Long multiline
 #     Description
-#     ''' (Bool, default = true)
+#     ''')
 #
-#   arg src 'Source'
-#   arg src 'Dest'
-#   arg times 'Foo' (Int)
+#   arg src (help='Source')
+#   arg dest (help='Dest')
+#   arg times (help='Foo')
+#
+#   rest files
 # }
 #
-# var opt, i = parseArgs(spec, ARGV)
+# var args = parseArgs(spec, ARGV)
 #
-# echo "Verbose $[opt.verbose]"
+# echo "Verbose $[args.verbose]"
 
 # TODO: See list
 # - It would be nice to keep `flag` and `arg` private, injecting them into the
```

Comment:
Not related to this PR, but just reminding myself that we discussed the flag/arg namespace issue here

https://oilshell.zulipchat.com/#narrow/stream/384942-language-design/topic/Flag.20parsing.20words.20.2F.20.22dialect.22.20scope.20-.20namespace.2C.20eval